### PR TITLE
TEC-4016 Simplifying the block editor checks and controls.

### PR DESCRIPTION
**Pull request by Camwyn**
Requires: https://github.com/the-events-calendar/tribe-common/pull/1661

 Deprecates the Gutenberg plugin compatibility and utilizes the new functionality in Common.

[TEC-4016]

[TEC-4016]: https://theeventscalendar.atlassian.net/browse/TEC-4016

### CODE REVIEW

File: src/Tribe/Editor/Compatibility.php [hunk 1]
```diff
@@ -6,6 +6,15 @@
...
 class Tribe__Events__Editor__Compatibility {
 
+	/**
+	 * Key we store the toggle under in the tribe_events_calendar_options array.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $blocks_editor_key = 'toggle_blocks_editor';
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility.php:16]: Use static.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/Objects/Event.php [hunk 2]
```diff
@@ -59,6 +60,20 @@ public function data( $key = null, $default = null ) {
...
+
+			/**
+			 * Grabs the tribe-start-date query param from the url if it exists
+			 * and if its a valid date then adds it to the global window object.
+			 */
+			if ( $start_date ) {
+				$start_date = Dates::build_date_object( $start_date, null, false );
+
+				if ( $start_date ) {
+					$this->data['tribe_start_date'] = $start_date->format( Dates::DBDATEFORMAT );
```
Comment (borkweb) [inline src/Tribe/Editor/Objects/Event.php:73]: Should this be `Dates::DBDATETIMEFORMAT` or will it always be just year, month, day?

File: src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+		// Standard clauses to include so all events except for the selected ones to hide will be included.
+		$clauses[] = " {$this->alias}.meta_value = '' ";
+		$clauses[] = " {$this->alias}.meta_value IS NULL ";
+
+		if ( is_array( $this->currentValue ) ) {
+			$event_status_ids = implode( ',', array_map( 'esc_sql', $this->currentValue ) );
+		} else {
+			$event_status_ids = "'" . esc_sql( (string) $this->currentValue ) . "'";
+		}
```
Comment (borkweb) [inline src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php:187]: The `array_map` approach is missing the quoting of the strings. As an aside, I think this whole block can be simplified a bit:

```suggestion
		if ( ! empty( $this->currentValue ) ) {
			$event_status_ids = implode( "','", array_map( 'esc_sql', (array) $this->currentValue ) );
		}
```

File: src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+		$hide_clauses[] = $wpdb->prepare(
+			" {$this->alias}.meta_value NOT IN (%s) ",
+			$event_status_ids,
+		);
```
Comment (borkweb) [inline src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php:192]: We should probably ensure that `$event_status_ids` holds something before we potentially add an empty value inside the `()`.

```suggestion
		if ( ! empty( $event_status_ids ) ) {
			$hide_clauses[] = $wpdb->prepare(
				" {$this->alias}.meta_value NOT IN (%s) ",
				$event_status_ids,
			);
		}
```

File: src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+		 * @param array<string|string>        $hide_clauses  The hide clauses on whether to hide canceled and postponed events.
+		 * @param array<string|string>        $clauses       The standard clauses to get all events.
+		 */
+		$where_clause = apply_filters( 'tec_event_status_filterbar_where_clause', '', $this->currentValue, $this->alias, $hide_clauses, $clauses );
+		if ( $where_clause ) {
+			$this->whereClause = $where_clause;
+
+			return;
+		}
+
```
Comment (borkweb) [inline src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php:211]:
```suggestion
```

**Review by borkweb — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/Objects/Event.php (hunk 2 above)
Comment (Camwyn) [reply src/Tribe/Editor/Objects/Event.php:73]: Looks like that was intentional: https://github.com/the-events-calendar/the-events-calendar/commit/f56e491329dc5d48d87605358c752ae5fab580dd

**Pull request closed without merging**